### PR TITLE
Review and remove deprecated properties and events from the grids

### LINKED ISSUE #10617: Deprecate rowID and rowData in interfaces
Following the removal of `rowID` and `rowData` properties from `IgxRowDirective` and all public classes implementing the `RowType` interface, considering the following:

Deprecate the old `rowID` and `rowData` from any or all of the following:

- [x] context object of [`rowSelectorTemplate`](https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html#L18)
- [x] [`IGridEditDoneEventArgs`](https://github.com/IgniteUI/igniteui-angular/blob/13d82c111d668dfdc8c0f4ec9e1498d0d3e32667/projects/igniteui-angular/src/lib/grids/common/events.ts#L20)
- [x] [`IRowToggleEventArgs`](https://github.com/IgniteUI/igniteui-angular/blob/13d82c111d668dfdc8c0f4ec9e1498d0d3e32667/projects/igniteui-angular/src/lib/grids/common/events.ts#L151)
- [x] [ `IPinRowEventArgs`](https://github.com/IgniteUI/igniteui-angular/blob/13d82c111d668dfdc8c0f4ec9e1498d0d3e32667/projects/igniteui-angular/src/lib/grids/common/events.ts#L165)
- [x] #13425
- [x] [`IPathSegment`](https://github.com/IgniteUI/igniteui-angular/blob/13d82c111d668dfdc8c0f4ec9e1498d0d3e32667/projects/igniteui-angular/src/lib/grids/common/grid.interface.ts#L42)

and introduce the new `key` or `primaryKey` (already done with https://github.com/IgniteUI/igniteui-angular/pull/12394) and `data`

**More stuff to be addressed in this PR:**

1) For better readibility and maintenance, destructure `IGridEditDoneEventArgs` to `ICellEditDoneEventArgs`
and `IRowEditDoneEventArgs` as follows:

 
`IGridEditDoneEventArgs` to:

- ICellEditDoneEventArgs
- IRowEditDoneEventArgs

`IGridEditEventArgs` to

- ICellEditEventArgs
- IRowEditEventArgs

Grid,interface.ts shoud look like:

    cellEditEnter: EventEmitter<ICellEditEventArgs>;
    cellEdit: EventEmitter<ICellEditEventArgs>;
    cellEditDone: EventEmitter<ICellEditDoneEventArgs>;
    cellEditExit: EventEmitter<ICellEditDoneEventArgs>;
    rowEditEnter: EventEmitter<IRowEditEventArgs>;
    rowEdit: EventEmitter<IRowEditEventArgs>;
    rowEditDone: EventEmitter<IRowEditDoneEventArgs>;
    rowEditExit: EventEmitter<IRowEditDoneEventArgs>;

 
DONE https://github.com/IgniteUI/igniteui-angular/pull/12832/commits/b92d8ff78c8b518013e9db77a3e238e9ea184d85?show-viewed-files=true&file-filters%5B%5D=

2) For `rowAdd` and `rowDelete` emit new interface, which is cancellable - maybe `IRowDataEventArgs` is the proper candidate.
2.1 - see comment on line 555 in grid,interface about this change - in the end we can reconsider if this will be introduced with 16.1 or other more major version
2.2 - add rowDelete to GridType, build correct object to be emitted

3) Change `rowAdd` and `rowDelete` types for treeGrid and Pivot to be compatible with GridType
4) The `primaryKey` argument also crept into add and edit events, not only `rowDeleted`:

- see changes in crud.service.ts and events.ts here https://github.com/IgniteUI/igniteui-angular/pull/12394/files
- primaryKey should not exist in EDIT events interface
- for rowAdd - is subject of change. Should consider this https://github.com/IgniteUI/igniteui-angular/issues/12091

5) The rest of the events, which emitted `rowID` until now - see if needed to migrate `rowID` to `rowKey` or `key`. For row events, it makes sense to be `key`, while for cell events it makes sense to be `rowKey`. Still, if cell event to emit `rowKey` in difference to row events, which will be emitting `key`, is a big effort, keep cell events to also emit `key` and make sure it is properly documented, so that a user knows what is reading through `eventArgs.key`

6) Make a list of the breaking changes and think of way to avoid them. Damyan can help here too.

### PR DESCRIPTION
**Pull request by ddincheva**
Closes #10617

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 1]
```diff
@@ -142,8 +142,9 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
             return;
         }
         const args = cell.createEditEventArgs(true);
-
-        this.grid.summaryService.clearSummaryCache(args);
+        if (!this.grid.crudService.row) {
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/grids/api.service.ts:145]: I feel the whole if may be excessive, may be is worth trying of any test will fail after removing this.

**Comment by hanastasov** [comment]
https://github.com/IgniteUI/igniteui-angular/blob/13d82c111d668dfdc8c0f4ec9e1498d0d3e32667/projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts#L48 still uses `rowID`

**Comment by radomirchev** [comment]
@hanastasov what is the status of the PR?

**Comment by radomirchev** [comment]
@hanastasov it was not discussed at the planning meeting, though you can work on this only if you have the availability. The priority remains low.

**Comment by kdinev** [comment]
@hanastasov @ddincheva Let's revive this for `16.1.x`

**Comment by ddincheva** [comment]
ALl the changes related to the removal of the deprecated grid properties have been already merged with this PR:  #13004

**Comment by ddincheva** [comment]
Closing this PR because the rest of the necessary changes related to the deprecated properties of the event's arguments will be introduced directly in version 16.1.0  with this PR #13427

**Pull request closed without merging**